### PR TITLE
Support crashtests in testfiles.affected_testfiles

### PR DESCRIPTION
**Pull request by stephenmcgruer**

### CODE REVIEW

File: tools/wpt/tests/test_testfiles.py [hunk 1]
```diff
@@ -11,3 +12,32 @@ def test_getrevish_kwarg():
...
+        "paths": {
+            "a/b/c/foo-crash.html": [
+                "acdefgh123456",
+                "crashtest",
+            ]
+        },
+        "url_base": "/",
+        "version": 7,
+    }
+    mfst = manifest.Manifest.from_json("/", manifest_json)
```
Comment (stephenmcgruer) [inline tools/wpt/tests/test_testfiles.py]: I am very interested in a better name than 'mfst' -_-

File: tools/wpt/tests/test_testfiles.py [hunk 2]
```diff
@@ -11,3 +12,32 @@ def test_getrevish_kwarg():
...
+    with patch("tools.wpt.testfiles.load_manifest", return_value=mfst):
+        # Dependent affected tests are determined by walking the filesystem,
```
Comment (stephenmcgruer) [inline tools/wpt/tests/test_testfiles.py]: I briefly tried both factoring out the `os.walk` from `testfiles.affected_testfiles` and also using `mkdtmp`/etc to build an entire file structure, but both seemed pretty painful. If we really want to do one of those (or have some other idea) I'm open to it.

File: tools/wpt/tests/test_testfiles.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
+import os.path
 from mock import patch
 
+from tools.manifest import manifest
```
Comment (Hexcles) [inline tools/wpt/tests/test_testfiles.py]:
```suggestion
from tools.manifest.manifest import Manifest
```

File: tools/wpt/tests/test_testfiles.py (hunk 1 above)
Comment (Hexcles) [reply tools/wpt/tests/test_testfiles.py]:
```suggestion
    manifest = Manifest.from_json("/", manifest_json)
```

File: tools/wpt/tests/test_testfiles.py [hunk 4]
```diff
@@ -11,3 +13,33 @@ def test_getrevish_kwarg():
...
+            "a/b/c/foo-crash.html": [
+                "acdefgh123456",
+                "crashtest",
+            ]
+        },
+        "url_base": "/",
+        "version": 7,
+    }
+    mfst = manifest.Manifest.from_json("/", manifest_json)
+    with patch("tools.wpt.testfiles.load_manifest", return_value=mfst):
```
Comment (Hexcles) [inline tools/wpt/tests/test_testfiles.py]:
```suggestion
    with patch("tools.wpt.testfiles.load_manifest", return_value=manifest):
```

**Review by Hexcles — APPROVED** [review]
I misread it as `msft` at first, but `mfst` is also a bad name...

The test setup is a bit convoluted but I agree with you that this is probably the most reasonable way for now.

**Comment by stephenmcgruer** [comment]
The 3.8 failure appears to be a network error in TC. Retrying.

**Pull request merged**